### ISSUE TITLE
Display 2 spaces instead of 4 without changing the code or project settings

### ISSUE THREAD

**Issue by seahindeniz** [microsoft/vscode#132776]
Hi, I'm working on projects that requires 4 space indentation, but I find this rule is a bit irritating. Is it possible to have an option to change the displaying/rendering behavior to show 4 spaces as 2 space without actually changing the spaces in the code. Such functionality would also need to work when I want to create a new file with projects indentation requirements, but would still render 4 spaces as 2.

**Comment by ceoshikhar** [comment]
@seahindeniz  @buondevid Checkout https://marketplace.visualstudio.com/items?itemName=kylepaulsen.stretchy-spaces, might be helpful.

**Comment by seahindeniz** [comment]
Hey @ceoshikhar thanks for the suggestion!
I just installed it and gave it a try **however**, the extension has some issues with guidelines, and I assume this is because of the insufficient API support.

The extension isn't compatible with built-in guidelines, so it suggests [another extension](https://marketplace.visualstudio.com/items?itemName=spywhere.guides) which has noticeable bugs.
  - When you scroll down, guidelines doesn't get re-render. You have to click in order to trigger rendering. I've opened an issue: https://github.com/spywhere/vscode-guides/issues/67
  - Empty lines don't get guidelines rendered due to some restriction, which is not good and doesn't look organic

Better yet, I would like this feature to be added to VSCode as a built-in feature, rather than having an extension that depends on APIs that doesn't give full support to line rendering functionality

**Comment by dsherret** [comment]
This issue would improve accessibility as it would allow people to more easily view repos that use spaces. There is an issue in prettier about switching to tabs by default (https://github.com/prettier/prettier/issues/7475), but it's not a real fix for the accessibility concerns compared to something like this.

**Comment by xenago** [comment]
A suggestion like this is an *actual fix* for accessibility concerns, unlike suggestions elsewhere to change entirecodebases over! Can we get this added in such a way so that x-spaces can be folded into y-spaces or a tab in a configurable manner? I think many braille users would appreciate being able to fold 4 spaces into a tab, for example.

**Comment by ljharb** [comment]
It would indeed be great for viewers of a repo that has chosen spaces not to have to suffer from that choice - the best choice is that they use tabs, but this sounds second-best.

**Comment by MattiasMartens** [comment]
i would like to see this too.

i am a newbie to ligatures, but can a ligature solve this? i.e. render 4 spaces to a ligature of width 2.

**Comment by ArthurBrussee** [comment]
Another +1 to this. Everyone has their own preference for 2/4/3(??) sized indents. Reading all code at work, home, and various languages at the same indent size could really help to pattern detect quickly and ease in to new languages. With the recent work to make spaces act as tabs as much as possible, it seems this might be really close!

**Comment by SandraSaliger** [comment]
fully agree! same request here!

**Comment by sjors-bootmine** [comment]
bump!

**Comment by ViewableGravy** [comment]
+1. My company has coding guidelines that require indentation to be 4 spaces but I prefer 2 spaces. Simply being able to visually change this without conflicting with the codebase would be amazing!

**Comment by jjjuk** [comment]
Yes people, I need this one like an air. I've got brain damaged from 4 tabspaces today. Wish me better mental health.

**Comment by foolo** [comment]
Darn, if only there were a special character that could change width depending on the user's preference :)

**Comment by AndreiRegiani** [comment]
Temporary workaround: https://github.com/kylepaulsen/vscode-stretchy-spaces
(It's buggy)

It would be fantastic for this feature to be natively supported.
Is there a configuration for this am I missing?

I'm editing a 2-space codebase but I'd prefer to render as 4-spaces without affecting the source code file.

**Comment by Mateo-EBA** [comment]
+1 !
Letting each user see code however they prefer without modifying the actual file is always a plus

**Comment by grumpyguvner** [comment]
+1 🙏🏻 this is a massive bummer for me too

**Comment by imluyu** [comment]
+1

**Comment by siegelaaron94** [comment]
Alice (Python dev):
Ugh, two-space indentation? Four spaces is the standard!

Ben (JavaScript dev):
Four is overkill. Two is cleaner and compact.

Charlie (voice of reason):
You know this debate was solved decades ago, right? Just use tabs. One per indent level. Everyone sets their own width in their editor. No arguments. No messy diffs.

Because we seem to have forgotten the ~70s-80s this would be a nice feature

**Comment by magnetikonline** [comment]
> You know this debate was solved decades ago, right? Just use tabs.

I'm with you 100% @siegelaaron94 - I'm stubborn, tabs is the better option - languages like Golang got this right.

But... I also author quite a bit of Python, and really like the use of https://github.com/psf/black (it's much like `go fmt`) - and enforces four space indents - would _love_ the ability to render these indents as two spaces to allow nicer side-by-side use of my editors.
